modules/tinymce/src/core/main/ts/api/util/I18n.ts

```diff
@@ -103,7 +117,7 @@ const translate = (text: Untranslated): TranslatedString => {
   const getLangData = (text: Untranslated) => {
     // make sure we work on a string and return a string
     const textstr = toString(text);
-    return Obj.get(langData, textstr.toLowerCase()).map(toString).getOr(textstr);
+    return Obj.get(langData, textstr).or(Obj.get(langData, textstr.toLowerCase())).map(toString).getOr(textstr);
```

Comment:
I know this is my code, but it just occurred to me that because JS isn't a lazy language this will double our calls to `Obj.get`. We might need to unroll this (`Obj.get` is just an `if` block with `Obj.has`) to avoid impacting startup performance (when we will hit this function a _lot_).

Reply:
That's a good point, is something like this what you had in mind?
```
return Obj.get(langData, Obj.has(langData, textStr) ? textStr : textStr.toLowerCase()).map(toString).getOr(textStr);

```

Reply:
It's not quite what I was after - `Obj.get` will call `Obj.has` again - but it's fast enough. `hasOwnProperty` calls should be cheap.

Reply:
Would this be the performant unroll you had in mind? Doing it this way means we don't have to create an Optional each time as well.
```ts
    return Obj.has(langData, textStr)
      ? langData[textStr].toString()
      : Obj.get(langData, textStr.toLowerCase()).map(toString).getOr(textStr);
```

To further improve performance we can even get rid of the need to create an Optional as well:
```
    if (Obj.has(langData, textStr)) {
      return langData[textStr].toString();
    } else {
      const lcTextStr = textStr.toLowerCase();
      if (Obj.has(langData, lcTextStr)) {
        return langData[lcTextStr].toString();
      } else {
      return textStr;
      }
    }
```

Thoughts?

Reply:
Sorry for not following up earlier. I don't quite remember exactly what I had planned, but the unrolled optional seems like the best answer (this might become a hot codepath with a lot of plugins running).

Do note that `toString` is a function, here, not just calling `.toString()` on the value. I don't know why it's using a custom toString, but I'd rather not investigate that right now 😂